src/commcare_cloud/ansible/plugins/inventory/csv.py

```diff
@@ -0,0 +1,195 @@
+
+        try:
+            if self.loader:
+                (b_data, private) = self.loader._get_file_contents(path)
+            else:
+                b_path = to_bytes(path, errors='surrogate_or_strict')
+                with open(b_path, 'rb') as fh:
+                    b_data = fh.read()
+
+            # Faster to do to_text once on a long string than many
+            # times on smaller strings
+            data = to_text(b_data, errors='surrogate_or_strict').splitlines()
+
+            self._parse(data)
+        except Exception as e:
+            raise AnsibleParserError(e)
+
+    def _parse(self, lines):
+        row_groups = self._parse_row_groups(lines)
+        for row_group in row_groups:
+            rows = list(csv.DictReader(row_group))
+            if 'hostname' in rows[0]:
+                self._parse_hosts(rows)
+            elif 'group' in rows[0]:
+                self._parse_groups(rows)
+
+    def _parse_hosts(self, rows):
+        for row in rows:
+            groups = self._get_host_groups(row)
+            host_vars = self._get_host_vars(row)
+            host = row['host_address']
+            host_group = row['hostname']
+            self.inventory.add_group(host_group)
+            self.inventory.add_host(host, host_group)
+            for group in groups:
+                self.inventory.add_group(group)
+                self.inventory.add_child(group, host_group)
+
+            self.populate_host_vars([host], host_vars)
+
+    def _get_host_groups(self, row):
+        return [
+            val.strip() for key, val in row.items()
+            if key.startswith('group') and val.strip()
+        ]
+
+    def _get_host_vars(self, row):
+        vars = {'hostname': row['hostname']}
+        for key, val in row.items():
+            if 'var' in key and val:
+                item_type, name = key.split('.') if '.' in key else ('S', key)
+                name = name.split(' ')[1]
+                vars[name] = conv_str2value(item_type, val)
+        return vars
+
+    def _parse_groups(self, rows):
+        for row in rows:
+            group = row['group']
+            self.inventory.add_group(group)
+            var_name, type_, raw_val = row['var'], row['type'], row['val']
+            self.inventory.set_variable(group, var_name, conv_str2value(type_, raw_val))
+
+    def _parse_row_groups(self, csv_lines):
+        """Parse CSV lines into groups each with their own header column"""
+        row_groups = []
+        current_group = []
+        for line in csv_lines:
+            if line.startswith('#'):
+                continue
+            if not line.strip() or line.startswith(','):
+                if current_group:
+                    row_groups.append(current_group)
+                    current_group = []
+            else:
+                current_group.append(line)
+        if current_group:
+            row_groups.append(current_group)
+        return row_groups
+
+
+def conv_str2value(item_type, item):
```

Comment:
The `ini` parser uses [ast.literal_eval](https://docs.python.org/2/library/ast.html#ast.literal_eval) to parse var values - I could switch to that but thought that being explicit was better. Open to opinions.